File: pawtracker-backend/src/main/java/com/pawtracker/entities/UserColony.java
package com.pawtracker.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "user_colonies")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserColony {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "colony_id")
    private Colony colony;

    @Column(name = "active_feeding")
    private boolean activeFeeding;

}

File: PawTrackerColonies-frontend/src/models/UserColony.ts
import { Cat } from "./Cat";
import { ColonyLocation } from "./Location";

interface UserColony {
    id: string;
    userId: string;
    colonyId: string;
    name: string;
    cats: Cat[];
    location: ColonyLocation;
    activeFeeding: boolean;
}

export default UserColony;
